zerver/lib/actions.py

```diff
@@ -4609,7 +4609,7 @@ def do_delete_messages(realm: Realm, messages: Iterable[Message]) -> None:
         send_event(realm, event, users_to_notify)
 
 def do_delete_messages_by_sender(user: UserProfile) -> None:
-    message_ids = Message.objects.filter(sender=user).values_list('id', flat=True).order_by('id')
+    message_ids = list(Message.objects.filter(sender=user).values_list('id', flat=True).order_by('id'))
```

Comment:
@mateuszmandera should confirm, but I think do actually want this to be a QuerySet, given how `move_messages_to_archive` does chunking?  Not sure.

Reply:
`list` seems right, chunking won't benefit from having a QuerySet - as of now, move_messages_to_archive will just end up executing the entire query to get a list of ids anyway.

Reply:
I guess the total messages sent by an individual user will be rarely over a few 100Ks, so we don't need to worry about memory issues for fetching a complete list of their IDs to the client.  OK.